docs/source/design/dataset_argument.rst

```diff
@@ -0,0 +1,63 @@
+.. -*- mode: rst -*-
+.. vi: set ft=rst sts=4 ts=4 sw=4 et tw=79:
+
+.. _chap_design_dataset_argument:
+
+********************
+``dataset`` argument
+********************
+
+.. topic:: Specification scope and status
+
+   This specification describes the current implementation.
+
+All commands which operate on datasets have a ``dataset`` argument (``-d`` or
+``--dataset`` for the :term:`CLI`) to identify a single dataset as the
+If ``--dataset`` argument is not provided, the context of an operation is operation-specific (e.g., for `clone` command -- the dataset which is being cloned), but typically - a dataset which current working directory belongs to.
```

Comment:
and crap -- I have missed that my suggestion killed the ending of prior sentence, sorry @mih.
In either case, at least this one better be split into multiple sentences, e.g.
```suggestion
context of an operation.
If ``--dataset`` argument is not provided, the context of an operation is operation-specific.
For example, `clone` command will consider the :term:`dataset` which is being cloned to be the context.
But typically, a dataset which current working directory belongs to is the context of an operation.
```

Reply:
I have adjusted your suggestion: `s/operation-specific/command-specific/` will commit now.